src/core/ActionAtomistic.cpp

```diff
@@ -77,7 +77,8 @@ void ActionAtomistic::requestAtoms(const vector<AtomNumber> & a) {
 // only real atoms are requested to lower level Atoms class
     else unique.insert(indexes[i]);
   }
-
+  updateUniqueLocal();
+  atoms.unique.clear();
```

Comment:
I see, so you removed the clear at every step, that's why you added several more clear around in the code. You say "neighbour list update" but it is `requestAtoms` right? Since in principle someone might implement his/her own criteria to request a new list of atoms